### PR TITLE
[Enhancement] Check parquet statistics correctness using sort order

### PR DESCRIPTION
**Pull request by letian-jiang**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 16 / 17 (94.12%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/formats/parquet/file_reader.cpp|16|17|94.12%|[419]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 5 / 10 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/formats/parquet/metadata.h|1|4|25.00%|[26, 27, 37]|
|:large_blue_circle:|src/formats/parquet/metadata.cpp|2|4|50.00%|[32, 33]|
|:large_blue_circle:|src/formats/parquet/file_reader.cpp|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 5 / 10 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/formats/parquet/metadata.h|1|4|25.00%|[26, 27, 37]|
|:large_blue_circle:|src/formats/parquet/metadata.cpp|2|4|50.00%|[32, 33]|
|:large_blue_circle:|src/formats/parquet/file_reader.cpp|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 193 / 214 (90.19%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/formats/parquet/metadata.h|1|2|50.00%|[58]|
|:large_blue_circle:|src/formats/parquet/metadata.cpp|188|208|90.38%|[48, 49, 50, 63, 68, 69, 130, 209, 212, 215, 259, 296, 383, 384, 385, 396, 414, 418, 454, 455]|
|:large_blue_circle:|src/formats/parquet/file_reader.cpp|4|4|100.00%|[]|

**Review by zombee0 — APPROVED** [review]
It will be better to check how parquet-mr works

File: be/src/formats/parquet/metadata.cpp [hunk 1]
```diff
@@ -35,4 +40,420 @@ std::string FileMetaData::debug_string() const {
...
+    if (VersionLt(ApplicationVersion::PARQUET_251_FIXED_VERSION())) {
+        return false;
+    }
+
+    return true;
+}
+
+// reference both be/src/formats/parquet/column_converter.cpp
+// and https://github.com/apache/parquet-format/blob/master/LogicalTypes.md
+SortOrder sort_order_of_logical_type(LogicalType type) {
```
Comment (letian-jiang) [inline be/src/formats/parquet/metadata.cpp:431]: very meat of this PR.

Comment (Smith-Cruise) [inline be/src/formats/parquet/metadata.cpp:431]: Can use as static function

**Review by Smith-Cruise — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**